tests/CMakeLists.txt

```diff
@@ -36,7 +36,8 @@ endif()
 # Compile options
 
 if (WIN32)
-    add_compile_options(-wd4100 -wd4514 -wd4668 -wd4710 -wd4711 -wd4820)
+    add_compile_options(-wd4100 -wd4514 -wd4625 -wd4626 -wd4668 -wd4710 -wd4711 -wd4774 -wd4820 -wd5025 -wd5026 -wd5027 -wd5039)
```

Comment:
nit: it would be nice that we add some comments about these warning message, such as 
// 5039 reported for nullptr argument